src/SyncTasks.ts

```diff
@@ -147,6 +147,13 @@ export interface Promise<T> extends Thenable<T>, Cancelable {
     
     // Defer the resolution of the then until the next event loop, simulating standard A+ promise behavior
     thenAsync<U>(successFunc: SuccessFunc<T, U>, errorFunc?: ErrorFunc<U>): Promise<U>;
+
+    // Turn this option on if you are debugging double resolve/success of this promise.
+    // In case of double resolve assert you will get two stacktraces - for the first resolve and
+    // the second. By default you would see only second resolve.
+    // Option adds extra overhead as on resolve Error object would be created, so it should be used
+    // with caution on release. Estimated overhead on mobile is around 0.5ms per error created on Nexus 5x android.
+    setTracingEnabled(enabled: boolean): Promise<T>;
```

Comment:
[This seems like a fine place for this addition. I wonder if we want a global option to overwrite this behavior: but we can add that if we need it.]